### PR TITLE
Initial controller support

### PR DESCRIPTION
**Pull request by MrPepperoni**
usage: main -d player1:controller:1
where the number after controller is the id of the controller, in range 0..15

starting main --calibrate prints all data read from the detected controllers

the keybindings for controllers can be customized in a settings.json file placed in the working directory of main

### CODE REVIEW

**Comment by MrPepperoni** [comment]
The controller mapping can be further specified by adding more complete mappings under the controller key. The most-specific found one will be used, where the key is
controlmappings.controller.${detected name of the controller}.${detected platform, can be "linux", "windows", "mac", "unknown"}.

File: src/input/controller_adapter_system.cpp [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+        // fill inputstate for player
+        is.mShoot = cs.getBool( "shoot" );
+        is.mShootAlt = cs.getBool( "shoot_alt" );
+        is.mUseNormalItem = cs.getBool( "use_normal" );
+        is.mReload = cs.getBool( "reload" );
+        is.mShowLeaderboard = cs.getBool( "show_leaderboard" );
+        is.mPause = cs.getBool( "pause" );
+        double vh = cs.getDouble( "view_horizontal" ), vv = cs.getDouble( "view_vertical" );
+        if( std::abs( vv ) >= 0.4 || std::abs( vh ) >= 0.4 )
```
Comment (HalalUr) [inline src/input/controller_adapter_system.cpp]: Settings.

File: src/input/controller_adapter_system.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        {
+            is.mOrientation = atan2( vv, vh );
+        }
+        Opt<Actor> actor = controlledPlayers[ controlledLocalPlayerId ];
+        if( actor.IsValid() )
+        {
+            Opt<IPositionComponent> actorPositionC = actor->Get<IPositionComponent>();
+            static double const radius = 400;
```
Comment (HalalUr) [inline src/input/controller_adapter_system.cpp]: Global Settings

Comment (MrPepperoni) [inline src/input/controller_adapter_system.cpp]: Donezo, for now all 3 (view_null_zone, move_null_zone, cursor_radius) under settings/controllers

**Pull request merged**